diff --git a/arch/arm64/boot/dts/qcom/msm8916-pm8916.dtsi b/arch/arm64/boot/dts/qcom/msm8916-pm8916.dtsi
index b1a7eafbee31..271d07b5a358 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-pm8916.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-pm8916.dtsi
@@ -145,7 +145,8 @@ pm8916_l12: l12 {
 
 		pm8916_l13: l13 {
 			regulator-min-microvolt = <3075000>;
-			regulator-max-microvolt = <3075000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-allow-set-load;
 		};
 
 		l14 {};
diff --git a/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi b/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
index c77ed04bb6c3..6f7adf6085df 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
@@ -46,6 +46,7 @@ leds {
 		led_r: led-r {
 			color = <LED_COLOR_ID_RED>;
 			default-state = "on";
+			linux,default-trigger = "heartbeat";
 			function = LED_FUNCTION_POWER;
 		};
 
@@ -58,6 +59,7 @@ led_g: led-g {
 		led_b: led-b {
 			color = <LED_COLOR_ID_BLUE>;
 			default-state = "off";
+			linux,default-trigger = "phy0tx";
 			function = LED_FUNCTION_WAN;
 		};
 	};
diff --git a/arch/arm64/boot/dts/qcom/msm8916.dtsi b/arch/arm64/boot/dts/qcom/msm8916.dtsi
index cb85f77cdc93..448ae7fdb2ec 100644
--- a/arch/arm64/boot/dts/qcom/msm8916.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916.dtsi
@@ -335,10 +335,10 @@ CLUSTER_PD: power-domain-cluster {
 		};
 	};
 
-	rpm: remoteproc {
-		compatible = "qcom,msm8916-rpm-proc", "qcom,rpm-proc";
+	smd {
+		compatible = "qcom,smd";
 
-		smd-edge {
+		rpm {
 			interrupts = <GIC_SPI 168 IRQ_TYPE_EDGE_RISING>;
 			mboxes = <&apcs 0>;
 			qcom,smd-edge = <15>;
